### ISSUE TITLE
当 label 中存在 ignore_index 时 CrossEntropyLoss 计算问题

### ISSUE THREAD

**Issue by QimingPeng** [PaddlePaddle/Paddle#48663]
### bug描述 Describe the Bug

直观 paddle 的 CrossEntropyLoss 在 reduction='mean' 时求 mean 的时候没有忽略 ignore_index

当没有 masked_lm_labels[:, 100:400] = -100 时，输出为
![f3ab2e3c1c4bcf31b68446eb51ef3d7a](https://user-images.githubusercontent.com/44738213/205256737-dfaa7f2e-cccb-40e4-a2df-bd8e78de98fd.png)

当有 masked_lm_labels[:, 100:400] = -100 时，输出为
![8518bac299789f6133dfc1df03b7aa6c](https://user-images.githubusercontent.com/44738213/205256762-7e046b32-974a-48b4-a6ab-5ef0b8cfb090.png)

loss_mlm_t 和 loss_mlm_p 分别为 torch 和 paddle 的 loss
看起来就像在 torch 的基础上乘了个 0.6 的系数，约等于 label 中 ignore_index 的比例

复现代码如下
```python
# paddlepaddle==2.3.2
# torch==1.11.0
import numpy as np
import paddle
lm_logits = np.random.uniform(-1, 1, size=[4, 512, 50265])
masked_lm_labels = np.random.randint(0, 50265, size=[4, 512])
# masked_lm_labels[:, 100:400] = -100
lm_logits_p = paddle.to_tensor(lm_logits)
masked_lm_labels_p = paddle.to_tensor(masked_lm_labels)

import torch
lm_logits_t = torch.tensor(lm_logits)
masked_lm_labels_t = torch.tensor(masked_lm_labels)

from torch.nn import CrossEntropyLoss as CrossEntropyLoss_t
loss_fct_t = CrossEntropyLoss_t(ignore_index=-100)
loss_mlm_t = loss_fct_t(lm_logits_t.view(-1, lm_logits_t.size(-1)), masked_lm_labels_t.view(-1))

from paddle.nn import CrossEntropyLoss
loss_fct_p = CrossEntropyLoss(ignore_index=-100)
loss_mlm_p = loss_fct_p(lm_logits_p.reshape((-1, lm_logits_p.shape[-1])), masked_lm_labels_p.reshape((-1, )))

print(loss_mlm_t)
print(loss_mlm_p)
```

**Comment by huangjun12** [comment]
是的，label中存在-100时，计算平均时会将其考虑进去。

```
print(loss_mlm_t)
print(loss_mlm_p*(512/(512-300)))
```

这样两者的值是一样的
